.circleci/config.yml

```diff
@@ -5,194 +5,98 @@
-    description: "Downloads & installs only the fresh copy of conda."
-    steps:
+              wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh;
+              chmod +x ~/miniconda.sh && ~/miniconda.sh -b;
+              echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV;
       - run:
-          name: setup conda afresh
+          name: Install apt packages
           command: |
-            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
-            chmod +x ~/miniconda.sh && ~/miniconda.sh -b
-            echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV
-      - run: conda create -n testenv
-
-  install_dependencies:
-    description: "Installs the necessary Python packages"
-    steps:
+              ./tools/circleci_dependencies_apt.sh
       - run:
           name: Install packages in conda env
           command: |
-            conda install -n testenv python=3.7 numpy scipy scikit-learn matplotlib pandas \
-            lxml mkl sphinx numpydoc pillow pandas -yq
-            conda install -n testenv nibabel sphinx-gallery junit-xml -c conda-forge -yq
-
-  build_docs:
-    description: "Installs Nilearn & builds documentation using Sphinx's make html-strict"
-    steps:
+              ./tools/circleci_dependencies.sh
       - run:
-          name: Building documentation
+          name: Find build type
           command: |
-            source activate testenv
-            pip install -e .
-            set -o pipefail && cd doc && make html-strict 2>&1 | tee log.txt
+              ./tools/circleci_build_type.sh
+      - run:
+          name: Verify build type
+          command: |
+              echo "PATTERN = $(cat pattern.txt)"
+              echo "BUILD = $(cat build.txt)"
+      - run:
+          name: build doc
+          command: |
+              source activate testenv;
+              echo "Conda active env = $CONDA_DEFAULT_ENV";
+              cd doc;
+              PATTERN=$(cat ../pattern.txt) make $(cat ../build.txt)
           no_output_timeout: 7h
-
-  store_results:
-    description: "Stores build times and artifacts"
-    steps:
       - store_test_results:
           path: doc/_build/test-results
       - store_artifacts:
           path: doc/_build/test-results
+          destination: test-results
+      - store_artifacts:
+          path: doc/_build/html/
+          destination: dev
       - store_artifacts:
-          path: doc/_build/html
+          path: doc/_build/html_stable
+          destination: stable
       - store_artifacts:
           path: coverage
       - store_artifacts:
           path: doc/log.txt
-
-  save_to_cache:
-    description: "Caches the downloaded packages & buit docs."
-    steps:
-      - save_cache:
-          key: v1-packages+datasets-{{ checksum "week_num" }}-{{ checksum ".circleci/packages-cache-timestamp" }}
-          paths:
-            - ../nilearn_data
```

Comment:
I think I'm missing something but I don't see the `nilearn_data` dir being cached anymore?

Reply:
You're absolutely right, I'm not caching `nilearn_data` at this point. 
I didn't do it in the first place for the following reasons:
- Prior to this PR, when we do a full build, we don't rely on caching (for the environment and the datasets), so we install everything from scratch and download all the data. I think this makes sense, so I kept this behavior. 
- When we do a partial build now, we only run the examples which were modified, so the amount of data that we have to download should be limited (assuming a PR doesn't modify lots of examples, which should be pretty rare...).

That being said, I agree that having the data cached could save time and avoid network connection failures, especially when having lots of active PRs (like in a sprint context...). I was therefore planning to restore it and split it into smaller caches (as you proposed in #2744) as this seems to be more efficient than having a huge cache. 

As for this PR, I can restore the full cache and split it later in a separate PR, or keep it like this and implement the splitted cache directly. Let me know what you think is best!

Reply:
ok that makes sense! let's do what you suggest then, remove caching here and in the next pr add the splitted cache. it will still be useful because even for a single example the download of some dataset can fail or take a long time